lib/x11utils.pm

```diff
@@ -68,7 +68,17 @@ all possible options should be handled within loop to get unlocked desktop
 =cut
 sub ensure_unlocked_desktop {
     my $counter = 10;
-    while ($counter--) {
+
+    # check if the screenbuffer was not updated and record a softfailure in that case
+    if check_screen 'screenlock' {
```

Comment:
Send the shift key if no screenlock is shown. (possible outdated screenbuffer)
I also realized that we should only apply the workaround if desktop is gnome.

```suggestion
    if (check_var('DESKTOP', 'gnome') && !check_screen('screenlock', 0) {
```

Reply:
This seems not logic to me, if we have a lockscreen and press shift we should not get another lockscreen